Source/Scene/LabelCollection.js

```diff
@@ -267,100 +270,136 @@ define([
-            } else if (verticalOrigin === VerticalOrigin.CENTER) {
-                glyphPixelOffset.y = -(maxY - dimensions.height) / 2 * scale - dimensions.descent * scale;
+            if (text.charAt(glyphIndex) === '\n') {
+                ++lineIndex;
+                lineOffsetY += lineSpacing;
+                lineWidth = lineWidths[lineIndex];
+                widthOffset = calculateWidthOffset(lineWidth, horizontalOrigin, backgroundPadding);
+                glyphPixelOffset.x = widthOffset * scale * resolutionScale;
             } else {
-                // VerticalOrigin.BOTTOM
-                glyphPixelOffset.y = (maxDescent - dimensions.descent + backgroundPadding.y) * scale;
-            }
+                glyph = glyphs[glyphIndex];
+                dimensions = glyph.dimensions;
+
+                if (verticalOrigin === VerticalOrigin.TOP) {
+                    glyphPixelOffset.y = dimensions.height - maxGlyphY - backgroundPadding.y;
+                } else if (verticalOrigin === VerticalOrigin.CENTER) {
+                    glyphPixelOffset.y = (otherLinesHeight + dimensions.height - maxGlyphY) / 2;
+                } else if (verticalOrigin === VerticalOrigin.BASELINE) {
+                    glyphPixelOffset.y = otherLinesHeight;
+                } else {
+                    // VerticalOrigin.BOTTOM
+                    glyphPixelOffset.y = otherLinesHeight + maxGlyphDescent + backgroundPadding.y;
+                }
+                glyphPixelOffset.y = (glyphPixelOffset.y - dimensions.descent - lineOffsetY) * scale * resolutionScale;
 
-            glyphPixelOffset.y *= resolutionScale;
+                if (defined(glyph.billboard)) {
+                    glyph.billboard._setTranslate(glyphPixelOffset);
+                }
 
-            if (defined(glyph.billboard)) {
-                glyph.billboard._setTranslate(glyphPixelOffset);
+                //Compute the next x offset taking into acocunt the kerning performed
+                //on both the current letter as well as the next letter to be drawn
+                //as well as any applied scale.
+                if (glyphIndex < glyphLength - 1) {
+                    var nextGlyph = glyphs[glyphIndex + 1];
+                    glyphPixelOffset.x += ((dimensions.width - dimensions.bounds.minx) + nextGlyph.dimensions.bounds.minx) * scale * resolutionScale;
+                }
             }
+        }
 
-            //Compute the next x offset taking into acocunt the kerning performed
-            //on both the current letter as well as the next letter to be drawn
-            //as well as any applied scale.
-            if (glyphIndex < glyphLength - 1) {
-                var nextGlyph = glyphs[glyphIndex + 1];
-                glyphPixelOffset.x += ((dimensions.width - dimensions.bounds.minx) + nextGlyph.dimensions.bounds.minx) * scale * resolutionScale;
+        if (defined(backgroundBillboard) && (text.split('\n').join('').length > 0)) {
```

Comment:
This too is part of `repositionAllGlyphs` and typically happens only when there is some substantial change made to the label.